xnat_admin_tools/remove_stale_data.py

```diff
@@ -36,11 +36,11 @@ def remove_stale(connection: pyxnat.Interface, days: int, dry_run: bool = False)
     for session in to_delete:
         typer.echo(f"Deleting {session}")
         p = connection.select.project(session["project"])
-        if not dry_run:
-            p.subject(session["subject_id"]).delete()
-            assert not p.subject(
-                session["subject_id"]
-            ).exists(), "Participant data was not deleted"
+        try:
+            if not dry_run:
```

Comment:
Neat pick, but not sure if it maters since if the code isn't executed, then there is no exception. It make more sense to me for `if` to go outside of the `try/catch`